src/accel/SharedParams.hh

```diff
@@ -84,19 +85,26 @@ class SharedParams
     //!@{
     //! \name Internal use only
 
+    using SPHitManager = std::shared_ptr<detail::HitManager>;
+    using SPOffloadWriter = std::shared_ptr<detail::OffloadWriter>;
+
     //! Hit manager, to be used only by LocalTransporter
     SPHitManager const& hit_manager() const { return hit_manager_; }
 
+    //! Optional offload writer, only for use by LocalTransporter
+    SPOffloadWriter const& offload_writer() const { return offload_writer_; }
+
     //!@}
 
   private:
     //// DATA ////
 
     std::shared_ptr<CoreParams> params_;
-    std::shared_ptr<detail::HitManager> hit_manager_;
+    SPHitManager hit_manager_;
     std::shared_ptr<StepCollector> step_collector_;
     VecG4ParticleDef particles_;
     std::string output_filename_;
+    std::shared_ptr<detail::OffloadWriter> offload_writer_;
```

Comment:
```suggestion
    SPOffloadWriter offload_writer_;
```

Reply:
Good point, we definitely want to do that! I didn't notice that property.